services/src/main/java/org/keycloak/keys/loader/ClientPublicKeyLoader.java

```diff
@@ -112,16 +113,18 @@ private static KeyWrapper getSignatureValidationKey(CertificateRepresentation ce
             kid = certInfo.getKid() != null ? certInfo.getKid() : KeyUtils.createKeyId(clientCert.getPublicKey());
             keyWrapper.setKid(kid);
             keyWrapper.setPublicKey(clientCert.getPublicKey());
-            keyWrapper.setType(clientCert.getPublicKey().getAlgorithm());
+            keyWrapper.setType(JavaAlgorithm.getKeyType(clientCert.getPublicKey().getAlgorithm()));
             keyWrapper.setCertificate(clientCert);
             keyWrapper.setIsDefaultClientCertificate(true);
+            keyWrapper.setCurve(clientCert.getPublicKey().getAlgorithm());
```

Comment:
It seems that `setCurve` makes sense if the key is regarding ECDSA or EdDSA.
In case of the key regarding RSA-SSA (RS, PS), it sets the value "RSA". If it does not cause a side-effect, ther is no problem. If causes, need to do some appropriate processing.

Reply:
Will be good to check this. I hope there is no side-effect with calling `setCurve` always, so we can call it in a way that is proposed in this PR and we don't need another `if` in the codebase to call it just for some algorithms. But leaving to @rmartinc to confirm this.

Reply:
This  `curve` property  was added by EdDSA (it's even not needed for ECDSA). It does no harm adding it for other keys (RSA for example) but we can just add it if the type is OKP (the only one that really needs or defined the curve). I'm sending a new PR now.